lib/engines/content_object_store/app/models/content_object_store/content_block_document.rb

```diff
@@ -2,4 +2,6 @@ class ContentObjectStore::ContentBlockDocument < ApplicationRecord
   has_many :content_block_editions,
            -> { order(created_at: :asc, id: :asc) },
            inverse_of: :content_block_document
+
+  attr_readonly :block_type
```

Comment:
I'm not sure how we could neatly fetch the content blocks from the publishing API and use that definition to scaffold our models and factories. The cost for that level of flexibility with meta programming and handling the  requests on load is likely to not be worth it.

I think a good start would be to have a single definition inside Whitehall. Then we only need to keep that and the schemas in sync? Rather than definitions being littered across Whitehall too.

Should we have a static file that we read in on load? Then we have a constant to use in our models as enums, factories, etc?

Reply:
We could lean on the GovukSchemas gem, but this would require us to pull in the schemas (probably at boot time) to a known location. It looks like this is only mainly used in test for most projects at the moment.

Reply:
For now then should we create our single known location within Whitehall, even if it's manual for now?

Reply:
I think I'd worry about it being manual. I'd like us to keep the link between the Publishing API schemas and Whitehall as explicit as possible. I'll have a think